vllm/utils.py

```diff
@@ -945,6 +945,8 @@ def flatten_2d_lists(lists: List[List[T]]) -> List[T]:
     return [item for sublist in lists for item in sublist]
 
 
+# TODO: This function can be removed if transformer_modules classes are
+# serialized by value when communicating between processes
```

Comment:
I think this should be removed, and properly tested with it's absence, as part of this PR's goals.

Reply:
That was my plan as well but I undid the removal due to [this comment](https://github.com/vllm-project/vllm/pull/6751#issuecomment-2395320126) from @youkaichao. It doesn't hurt to keep it in for now if it can help the PR get merged. The removal can be a quick follow-on PR where we can test edge cases.